lib/child_process.js

```diff
@@ -473,7 +473,7 @@ function spawnSync(/*file, args, options*/) {
 
   // Validate maxBuffer, if present.
   if (options.maxBuffer != null &&
-      !(Number.isInteger(options.maxBuffer) && options.maxBuffer >= 0)) {
+      !(typeof options.maxBuffer === 'number' && options.maxBuffer >= 0)) {
     throw new TypeError('"maxBuffer" must be an unsigned integer');
```

Comment:
That's a good point. The integer check is still required otherwise we'll have nonsense things like `0.1` accepted as the maxBuffer. The check should likely be something like:

```
if ((Number.isInteger(options.maxBuffer) || options.maxBuffer === Infinity) &&
    options.maxBuffer >= 0) {
  ...
}
```
And the error message should certainly be updated but that would upgrade this to a semver-major change